### PR TITLE
Add the hostDevice tests to the network operator project

### PR DESCRIPTION
**Pull request by abdallahyas**
Added two tests for the host device feature of the network-operator:
testing host device alone, and testing host device with the OFED.

### CODE REVIEW

File: nic_operator/nic_operator_ci_test.sh [hunk 1]
```diff
@@ -466,6 +489,99 @@ function test_predefined_name {
...
+    nic_policy_create "$sample_file"
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: error in creating the example nic_policy."
+        return $status
+    fi
+
+    sleep 10
+
+    test_rdma_single_pod "" "$MACVLAN_NETWORK_DEFAULT_NAME" "${resource_name_prefix}/${resource_name}"
```
Comment (e0ne) [inline nic_operator/nic_operator_ci_test.sh]: Why MACVLAN_NETWORK_DEFAULT_NAME?

File: nic_operator/nic_operator_ci_test.sh [hunk 2]
```diff
@@ -466,6 +489,99 @@ function test_predefined_name {
...
+    sleep 10
+
+    test_ofed_drivers
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: Ofed modules failed!"
+        return $status
+    fi
+
+    test_rdma_single_pod "" "$MACVLAN_NETWORK_DEFAULT_NAME" "${resource_name_prefix}/${resource_name}"
```
Comment (e0ne) [inline nic_operator/nic_operator_ci_test.sh]: Do we need MACVLAN_NETWORK_DEFAULT_NAME here?

**Review by e0ne — COMMENTED** [review]
Where is host device network defined?

File: nic_operator/nic_operator_ci_test.sh (hunk 1 above)
Comment (abdallahyas) [reply nic_operator/nic_operator_ci_test.sh]: Fixed it to use a hostdevice network

File: nic_operator/nic_operator_ci_test.sh (hunk 2 above)
Comment (abdallahyas) [reply nic_operator/nic_operator_ci_test.sh]: fixed it to use a hostdevice network

**Comment by abdallahyas** [comment]
> Where is host device network defined?

Added a function to prepare an example hostdevice network. please have a look

**Review by ykulazhenkov — APPROVED** [review]

**Pull request merged**